### DISCUSSION TITLE
CAD and rx sensitivity

### DISCUSSION THREAD

**Discussion by garudaonekh** [jgromes/RadioLib#1003] [Q&A]
Hi,
Does using CAD on SX1278 reduce its receiving sensitivity? I mean affecting its receiving range

I go through several post regarding using CAD affect the receiving sensitivity.

Regards;

**Comment by HeadBoffin** [comment]
It was explained to me as the electronics have to move between different spreading factors at a fast pace to ensure they can report back any activity so that you can then choose to try to hear the broadcast. If the radio is tuned to one set of parameters it is better placed to hear marginal signals as it gets more time to lock on to the preamble & correct for any frequency drift in the transmitter.

It can be a useful way to use point to point / mesh networks that may have nodes that are adjusting their parameters but mostly I set a schedule for transmission parameters.

What are you hoping to do with it?

**Comment by jgromes** [comment]
@garudaonekh I have not observed this effect personally, maybe this question would be better suited for the Semtech developer forum.

EDIT: Hmm, looks like the Semtech forum is now read-only. Hope that's just temporary ... https://forum.lora-developers.semtech.com/

![Screenshot_81](https://github.com/jgromes/RadioLib/assets/3407192/66082ce1-65aa-416b-b3b5-284610053180)

**Reply by HeadBoffin** [reply]
Looks like the recent spam may have pushed it in to the "not worth the bother" category!

**Reply by garudaonekh** [reply]
Yes, it's now read only

**Comment by garudaonekh** [comment]
I develop both transmitter and receiver and only 1 transmitter but many receivers. 
Transmitter is connected to main power line but Receiver will live on very small solar battery(300mah).

Currently, I set PreambleLength to 600 on the transmitter and let the receiver to sleep 2.5 seconds before doing CAD scanning. A lot of power is save,( it consume 1ma while 2.5 seconds sleeping). Around 2% package loss with this header error (RADIOLIB_ERR_LORA_HEADER_DAMAGED (-24)). Maybe my preamplelength and sleep calculation is wrong.

**Reply by jgromes** [reply]
>  I set PreambleLength to 600 on the transmitter

That might violate some duty cycle/dwell time regulations ...

> Maybe my preamplelength and sleep calculation is wrong.

Possibly, though you have not shared it so we cannot check.

> Around 2% package loss

If we're talking about real-world usage, then this seems like a reasonable number. Did you evaluate the RSSI/SNR on the recevier? Is there enough margin in the link budget?

**Reply by HeadBoffin** [reply]
2% is a very good loss rate considering what you are doing.

Why are you using CAD if you have one transmitter - you could just open a receive with a timeout. I've only used CAD for a many sensors transmitting to a single receiver that uses CAD to listen across a range of SFs.

**Reply by garudaonekh** [reply]
I use your SX127x_CAD_Receive code. But add sleep and delay()

   ```
if(!receiving) {
      ```
radio.sleep();
      delay(2500);
```
      int state = radio.startChannelScan();
      if (state != RADIOLIB_ERR_NONE) {
        Serial.print(F("[SX1278] Starting new scan failed, code "));
        Serial.println(state);
      }
    }
```
I use setPreamble length to 600.

Regarding RSSI/SNR evalutation, I haven't done yet.

**Reply by garudaonekh** [reply]
I  don't get your point about **"a receive with a timeout"**. My problem is all the receivers live on battery power

**Reply by HeadBoffin** [reply]
From what you describe, you are using CAD to detect a transmission which uses the battery. However if you have a schedule and you know what the frequency & SF is, you can just open the receive window which should be far more efficient - you can set the timeout, with a good RTC on the MCU you just set a timeout for about 1 second and if there is something to receive, it will take care of the rest of it for you.

**Reply by garudaonekh** [reply]
unfortunately, Receiver does not know when data will be sent.

**Reply by jgromes** [reply]
> Receiver does not know when data will be sent.

Is there any reason it couldn't? You can transmit e.g. every hour, and then the receivers "know" that there will be 60 minutes in between transmissions. You can sleep e.g. for 58 of those minutes, then start reception. If you receive a packet, you know that the next one is also 60 minutes away, so you can sleep for 60 more minutes. Or your data transmission can contain the timestamp of the next packet ... the point @HeadBoffin was making is that there are options that don't use CAD at all, and don't require huge preambles (which might be exceeding duty cycle and/or dwell time, and that can be illegal!).

In any case, to answer your original question: if you're worried about impact of CAD on receiver sensitivity, then you don't have to use CAD. I actually don't think there should be any impact (since you are receiving on a fixed SF), but I don't know.

**Reply by garudaonekh** [reply]
Data is sent by Relay switch which is controlled by user thus don't have regular interval.
